ui/jobs/jobs.js

```diff
@@ -2128,6 +2128,16 @@ class Bars {
     if (!this.o.gpBar) return;
     this.o.gpBar.value = this.gp;
     this.o.gpBar.maxvalue = this.maxGP;
+
+    // GP Alarm
+    if (this.gp < this.options.GpAlarmPoint) {
+      this.gpAlarmReady = true;
+    } else if (this.gpAlarmReady && this.gp >= this.options.GpAlarmPoint) {
```

Comment:
Re: potion.  Do you think it'd be possible to delay the gp alarm for 0.5-1 second, and cancel it if a cordial is used in the log?

Reply:
Hmmm. I don't know how to read log...
I was thinking about checking GP different.

`if (this.gp < this.options.GpAlarmPoint) {`
to
`if (this.gp < this.options.GpAlarmPoint && this.gp >= this.options.GpAlarmPoint - 20) {`

so `this.gpAlarmReady = true;` won't happen before gp gets close to target GP point.

What you'd said is logically better I think, but I don't know how to read log.

Reply:
jobs reads the log in `OnLogEvent`, for what it's worth.

Reply:
okay, I'll do some research tomorrow, I should sleep now. I'll do tomorrow.

Reply:
here's the regex I was using for detecting cordials when making a similar trigger for personal use, maybe it'll be helpful
`cordialRegex = new RegExp(e.detail.name + ':20(017FD|F5A3D|F844F|0420F|0317D):');`

Reply:
I've found the log, and I can make this, but I don't know if this log is differ by game client version. 
```
[18:22:06.761] 15:1008A40B:블러드트레일:200317D:Item_317D:1008A40B:블러드트레일:E:1900000:0:0:0:0:0:0:0:0:0:0:0:0:0:0:18831:18831:0:0:0:1000:0.2368442:-0.08807932:3.814697E-06:-3.141593:18831:18831:0:0:0:1000:0.2368442:-0.08807932:3.814697E-06:-3.141593:00000625
[18:22:07.000] 00:082c:블러드트레일이 고성능 강장제를 사용했습니다.
[18:22:07.000] 00:08ad:⇒ 블러드트레일이 GP를 400 회복했습니다.
```
I am thinking about opcode part, which changes in every patches, on second line `082c` and on thrid line `08ad` is opcode I guess.
but is there no opcode on the first line?
If so, I can make that log for international use.

Reply:
I've found two cordials id is `17FD` (NQ: 300 GP / HQ: 350GP) and `317D` (400 GP).
I think this trigger will happen when you eat food, too. I have never eat food for GP, but as far as I know, there is GP food.
```
configured GP alarm for 800
1. now GP : 790
2. eat food
3. now GP: 810
-> alarm rings
```
like this..
I don't think it's good to add all the food's id in regex. That makes developers add new ids on new patches.
So.... maybe make this problem just left in program.

Reply:
or if opcode has no problem, it would be better to add chat regex.
`블러드트레일이 GP를 400 회복했습니다.` This is Korean battle chat log with GP healed(?).
So, 
add 
```
this.GpPotionRegex = function() {
  return /(이|가) GP를 (\y{Float}) 회복했습니다./;
};
```
on `resources\lang\ko.js` (and other languages)
```
if (log.search(e.detail.name + gLang.GpPotionRegex()) >= 0) {
  cancelGpAlarm();
}
```
on `ui\jobs\jobs.js`